adr/recipes/config_durations.py

```diff
@@ -7,22 +7,54 @@
 """
 from __future__ import print_function, absolute_import
 
-from ..recipe import RecipeParser
-from ..query import run_query
+from ..recipe import get_argument_groups
+from ..recipe import execute_query
 
+QUERY_ARGUMENTS = [
+    "branch", "date",
+    {'sort-key': [
```

Comment:
Since these arguments aren't being re-used, there's no need to label them or store them in an extra list. Each argument here can be of the form:

    [['--sort-key'], {
       'type': int,
        ...
    }]

This does mean that you'll need to change the logic in `recipe.py` to extend the 'common' groups so that the final list looks like:

    [
        [['--arg1'], { ... },
        [['--arg2'], { ... },
        ...
    ]

Reply:
Yes I changed the logic following your guide.